### PR TITLE
Fix morph becoming invisible and unclickable if it disguises as an airlock

### LINKED ISSUE #3376: invisible morph disguises
## Reproduction:

Morphing/disguising as any open airlock turns you into a completely invisible, unclickable object. Most likely caused by sprite overlays being on a temporarily invisible object

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/96285574-invisible-morph-disguises?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

Fixes a bug that would cause the morph to become invisible and unclickable if it tries to use an airlock as an disguise.
Also fixes screens not showing in disguise for computers or apc's.
I basically just convert the icons of the vis_content list objects into overlays.
closes #3376

## Why It's Good For The Game

Invisible Morph that cannot be hit is bad

## Changelog
:cl:
fix: Fixes morph becoming invisible and unclickable if it tries to disguise as an airlock
fix: Fixes computer screen and apc screen not showing in disguise
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
Didn't know this happened and came to github to report it after accidentally shift clicking an airlock. Nice to see the fix is already up.

File: code/modules/antagonists/morph/morph.dm [hunk 1]
```diff
@@ -214,6 +214,11 @@
 					"<span class='notice'>You twist your body and assume the form of [target].</span>")
 	appearance = target.appearance
 	copy_overlays(target)
+	if(target.vis_contents)
+		for(var/I in 1 to length(target.vis_contents))
+			var/obj/p = target.vis_contents[I]
+			icon = p.icon
+			add_overlay(list(p.icon_state, p.overlays))
```
Comment (park66665) [inline code/modules/antagonists/morph/morph.dm]:
```suggestion
	copy_overlays(target, TRUE)
	if(length(target.vis_contents))
		for(var/atom/A as() in target.vis_contents)
			add_overlay(A)
			copy_overlays(A)
```

Ideally, we should control all `vis_contents` via `SSvis_ovevrlays` and copy `vis_contents` with it, but this deep appearance copying is currently a better solution.

Comment (MNarath1) [reply code/modules/antagonists/morph/morph.dm]: Yeah that looks better trough i do not think the second copy_overlays is needed as add_overlay(A) already copies the icon and the overlays of an object.
At least from what i found out trough testing.

Comment (MNarath1) [reply code/modules/antagonists/morph/morph.dm]: Also using copy_overlays(A) will override the list as it looks like and thus deleting all the overlays already applied resulting in the morph to become invisible again.

File: code/modules/antagonists/morph/morph.dm [hunk 2]
```diff
@@ -213,7 +213,10 @@
 	visible_message("<span class='warning'>[src] suddenly twists and changes shape, becoming a copy of [target]!</span>", \
 					"<span class='notice'>You twist your body and assume the form of [target].</span>")
 	appearance = target.appearance
-	copy_overlays(target)
+	copy_overlays(target, TRUE)
+	if(length(target.vis_contents))
+		for(var/atom/A as() in target.vis_contents)
+			add_overlay(A)
```
Comment (park66665) [inline code/modules/antagonists/morph/morph.dm]:
```suggestion
	if(length(target.vis_contents))
		add_overlay(target.vis_contents)
```

Internal BYOND magic: copying `appearance` also copies overlays (the reason why adding atom as an overlay also adds that atom's overlays)
Original code: hey let's copy overlays separately
Me: Surprised pikachu picture

Anyways, this should be the cleanest way to achieve this.

Comment (MNarath1) [reply code/modules/antagonists/morph/morph.dm]: Hmm i am sure i already tried something like that but yeah you are right it works can't argue with the result.

File: code/modules/antagonists/morph/morph.dm [hunk 3]
```diff
@@ -213,7 +213,9 @@
 	visible_message("<span class='warning'>[src] suddenly twists and changes shape, becoming a copy of [target]!</span>", \
 					"<span class='notice'>You twist your body and assume the form of [target].</span>")
 	appearance = target.appearance
-	copy_overlays(target)
+	copy_overlays(target, TRUE)
```
Comment (park66665) [inline code/modules/antagonists/morph/morph.dm]:
```suggestion
```

I told you, `appearance` copying above also copies overlays. The thing works without this line. I even tested it.

**Review by park66665 — APPROVED** [review]
All clear. This PR is good to go.

**Pull request merged**